### ISSUE TITLE
No defaults in enum switch

### ISSUE THREAD

**Issue by matrinox** [realm/SwiftLint#131]
I find the exhaustive `enum` warnings from the compiler super helpful. It ensures that whenever a new `enum` case is added, every `switch` statement is updated. But this won't work well with `default`. Often times the `default` is there as an easy way to group behaviour for the other cases at the time. I'd like to see this added as a warning

**Comment by kevindelord** [comment]
+1 we also make sure to never use the `default` case in our enums :)

**Comment by Noobish1** [comment]
I'd love this feature also.

**Comment by jpsim** [comment]
I think this is a good rule to have, but it would be challenging to detect when the variable being switched on is an enum.

**Comment by acecilia** [comment]
I would love to see this too :)

**Comment by valentinradu** [comment]
This would be great!

**Comment by sammy-SC** [comment]
@jpsim 

you mentioned that it would be challenging to detect when the variable being switched on is an enum.
In simple case like the one below, shouldn't it be sufficient to see whether the condition next to case is prepended with a dot? In this case `.one`? I think it should be safe to assume that variable being switched on is an enum if all conditions are prepended with a dot.

```swift
enum MyEnum {
  case one, two
}

let myEnum = MyEnum.two

switch myEnum {
  case .one: break
  ↓default: break
}
```

Please let me know whether you think this could work, I would be happy to implement this rule.
